spec/controllers/api/v2/footnotes_controller_spec.rb

```diff
@@ -1,108 +1,147 @@
-              geographical_area: {
-                data: {
+                {
                   id: String,
-                  type: 'geographical_area',
+                  type: 'chapter',
                 },
-              },
+              ],
             },
           },
-        ],
-        meta: {
-          pagination: {
-            page: Integer,
-            per_page: Integer,
-            total_count: Integer,
+        },
+      ],
+      included: [
+        {
+          id: String,
+          type: 'chapter',
+          attributes: {
+            goods_nomenclature_item_id: String,
+            description: String,
+            formatted_description: String,
+            producline_suffix: String,
           },
         },
-      }
+        {
+          id: String,
+          type: 'heading',
+          attributes: {
+            goods_nomenclature_item_id: String,
+            description: String,
+            formatted_description: String,
+            producline_suffix: String,
+          },
+        },
+        {
+          id: String,
+          type: 'heading',
+          attributes: {
+            goods_nomenclature_item_id: String,
+            description: String,
+            formatted_description: String,
+            producline_suffix: String,
+          },
+        },
+        {
+          id: String,
+          type: 'measure',
+          attributes: {
+            goods_nomenclature_item_id: String,
+            validity_start_date: String,
+            validity_end_date: String,
+          },
+          relationships: {
+            goods_nomenclature: {
+              data: {
+                id: String,
+                type: 'heading',
+              },
+            },
+            geographical_area: {
+              data: {
+                id: String,
+                type: 'geographical_area',
+              },
+            },
+          },
+        },
+      ],
+      meta: {
+        pagination: {
+          page: Integer,
+          per_page: Integer,
+          total_count: Integer,
+        },
+      },
+    }
+  end
+
+  before do
+    measure_with_goods_nomenclature = create(:measure, :with_base_regulation, goods_nomenclature: create(:heading, :with_description))
+    measure_no_goods_nomenclature = create(:measure, :with_base_regulation, goods_nomenclature: nil)
+
+    create(:footnote_association_goods_nomenclature, footnote:, goods_nomenclature: create(:heading, :with_description))
+    create(:footnote_association_goods_nomenclature, footnote:, goods_nomenclature: create(:chapter, :with_description))
+    create(:footnote_association_goods_nomenclature, footnote:, goods_nomenclature_sid: '9999') # Broken reference
+
+    create(:footnote_association_measure, measure: measure_with_goods_nomenclature, footnote:)
+    create(:footnote_association_measure, measure: measure_no_goods_nomenclature, footnote:)
+
+    Sidekiq::Testing.inline! do
+      TradeTariffBackend.cache_client.reindex
+      sleep(2)
```

Comment:
what's happen if we remove the sleep?
... it is inline, should be fine 🤔

Reply:
Is async AFAIK

Reply:
I'll try it

Reply:
Fails

Reply:
👍  thanks for trying 😄